### ISSUE TITLE
Make interface more keyboard friendly

### ISSUE THREAD

**Issue by schneis** [tannerhelland/PhotoDemon#277]
The program is really cool, but requires a mouse for even simple things.  Making better use of the keyboard would really improve usability.

One example - the Move & Resize tool ... select the Layer position X field, press Tab - the next control selected is the Layer size width up/down buttons (not Layer position Y input box which would make more sense).  Tabbing between the fields always selects the up/down arrows and not the numeric value making the keyboard almost useless here.

Another example - the menus do not all have hotkeys to select them.  I frequently use Effects -> Pixelate -> Crystallize and only Alt+C is available to open the Effects menu.  Being able to do something like Alt+C, P, Y to open Crystallize or being able to define my own hotkey (eg Ctrl+Y) would be very useful.

**Comment by tannerhelland** [comment]
Thank you for this excellent feedback, @schneis.  These kind of small details are incredibly easy for me to miss, and it's a huge help when users provide concrete usability feedback like you have.

There are several different issues at play here - some easy to solve, others more complicated - but I'll work on addressing all of them as quickly as I can.

First up: I've just committed a series of improvements to tab-key ordering.  PhotoDemon is built on a custom UI kit that - as you point out - heavily focuses on mouse usage.  Keyboard interactions still need some work!

Fixes I've just uploaded include...

- Spin controls (up/downs) now focus the numeric value by default, instead of the up/down buttons.
- Tab key order is usually determined automatically by the UI engine, using a left-to-right, top-to-bottom pattern.  On windows where this order doesn't make sense, I can now specify a custom order instead.  I've just implemented this on the Move/Size tool options panel.  If you encounter other panels where the default tab order is bad, please let me know and I'll write a custom order for them as well.
- I've just now realized that some controls do not convey focus well when navigating them via keyboard.  I'm going to start touching these up next, to ensure that the currently focused control is always obvious, whether using mouse *or* keyboard.

Hotkeys is a messier problem to solve, as it really needs a full-blown customization tool that allows users to set their own shortcuts.  (Trying to implement a default one is not easy, as PD is available in so many different locales.)

This has been on my to-do list for awhile but there are a lot of messy details I have yet to work out.  If you know of other software that handles this feature well, I'd love to investigate them to gain a better understanding of the work involved.

Thank you!

**Comment by tannerhelland** [comment]
Over a year later, I'm back with an update!  😱 

Many improvements have been made in this area over the past year, most of which I (unfortunately) forgot to tag against this commit.  For example, all menus are now navigable by keyboard, regardless of your current language - PD automatically generates these "mnemonics", as Windows calls them, at run-time.  So using the original example of "Crystallize", `Alt+C > P > R` now launches the Crystallize effect, and each menu hotkey is underlined when `Alt` is depressed so you don't have to guess.  (`R` is used for the `Crystallize` hotkey because `C` is already taken by `Color Halftone`, above it.) 

*(Also, the Crystallize filter is much faster and has tons of new features in nightly builds, so that's fun!)*

Tab key navigation is now available on **all** dialogs, too, and various UI elements now display keyboard focus much better.  (You can also do convenient things like move spin control values up/down using up/down keys, even if you're in the text section of the control.)

My last major keyboard-related fix is custom hotkeys, which is high on my to-do list.  I hope to have this available before the next major release.  Once this is ready for testing, I'll update it in its own issue since it's quite a large endeavor.

Thank you again for prompting me to take keyboard interactions more seriously, @schneis !  If anything in the current nightly builds remains unintuitive, please let me know and I'll get it fixed.
